dash/dash.py

```diff
@@ -217,6 +225,24 @@ def add_url(name, view_func, methods=('GET',)):
         self._cached_layout = None
         self.routes = []
 
+        # hot reload
+        self._reload_hash = _generate_hash() \
+            if hot_reload else ''
+        self._reload_interval = hot_reload_interval
+        self._hard_reload = False
+        self._lock = threading.RLock()
+        self._watch_thread = None
+        if hot_reload:
+            self._watch_thread = threading.Thread(
+                target=lambda: _watch.watch([self._assets_folder],
+                                            self._on_assets_change,
+                                            sleep_time=0.5))
```

Comment:
Shouldn't `sleep_time` be `hot_reload_interval / 1000`? I see that `hot_reload_interval` is passed to the front-end via `config`, but I'm not sure if it is intended to be the same on the back-end. If not (e.g. back-end should have a fixed constant refresh interval regardless of `hot_reload_interval` value) I think you should make this a constant set in the `_watch` file.

Reply:
The sleep time is the poll time for the assets watch thread and `hot_reload_interval` is for the client requests interval. I don't think the user will need to change the sleep time so it can be a constant.

Reply:
I don't think it matter much, it just the timer for when to run the check in the back end and should not be linked in any way to the request timer that can be changed. It should be a constant that work goods in the conditions. We don't want run that check too often because of the `GIL` and it should still be low so you can get the reload a short while after making changes. So I think a check every half a second is good with this in mind.

In my `asyncio` based implementation it was a really low number like `yield from asyncio.sleep(0.000001)` and it didn't make a difference in the performance of the app.